### ISSUE TITLE
Cannot use OpenPI → PyTorch Pi0 checkpoint with examples/rtc/eval_dataset.py (missing LeRobot policy artifacts)

### ISSUE THREAD

**Issue by khaledmohamed00** [huggingface/lerobot#2616]
Description

I have a fine-tuned Pi0 model trained with OpenPI (JAX) and converted to PyTorch using the official conversion script:

https://github.com/Physical-Intelligence/openpi/blob/main/examples/convert_jax_model_to_pytorch.py

The conversion script outputs a directory that contains:

```
model.safetensors

configs.json
```

where configs.json looks like:

```
{
  "action_dim": 32,
  "action_horizon": 10,
  "paligemma_variant": "gemma_2b",
  "action_expert_variant": "gemma_300m",
  "precision": "bfloat16"
}
```

I would like to evaluate this fine-tuned Pi0 model in LeRobot’s RTC pipeline, using:

```
python examples/rtc/eval_dataset.py \
  --policy.path=/path/to/o10_pytorch_checkpoint \
  --dataset.repo_id=HuggingFaceVLA/libero \
  --rtc.execution_horizon=10 \
  --rtc.max_guidance_weight=10.0 \
  --device=cuda
```

Here /path/to/o10_pytorch_checkpoint is the directory produced by the OpenPI convert_jax_model_to_pytorch.py script (with only model.safetensors + configs.json).

What LeRobot expects vs. what I have

From the RTC examples (examples/rtc/eval_dataset.py, examples/rtc/eval_with_real_robot.py), LeRobot expects --policy.path to point to a LeRobot policy directory with at least the following files:

```
config.json

model.safetensors

policy_preprocessor.json

policy_postprocessor.json

policy_preprocessor_step_5_normalizer_processor.safetensors

policy_postprocessor_step_0_unnormalizer_processor.safetensors
```

However, the OpenPI → PyTorch conversion only gives me:

```
model.safetensors

configs.json (minimal, not LeRobot’s config.json)
```

There is no:

LeRobot-style config.json (with policy type, feature specs, etc.)

preprocessor / postprocessor definitions

normalizer / unnormalizer safetensors

As a result, pointing --policy.path to this directory does not match what the RTC scripts expect from a “policy”.

Loading as PI0Policy directly

If I try to load the checkpoint programmatically:

```
from lerobot.policies.pi0.modeling_pi0 import PI0Policy
import torch

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
pretrained_policy_path = "/path/to/o10_pytorch_checkpoint"

policy = PI0Policy.from_pretrained(pretrained_policy_path).to(device)
```

I see logs like:

```
✓ Loaded state dict from model.safetensors

Remapped XXX state dict keys

warnings about some missing/unmapped keys
```

So model.safetensors itself can be read, but the overall directory is still not a valid LeRobot policy artifact that works with examples/rtc/eval_dataset.py and examples/rtc/eval_with_real_robot.py.

Problem in short

There is currently no supported / documented way to take a fine-tuned Pi0 model from OpenPI (JAX, converted to PyTorch) and turn it into a LeRobot-compatible policy directory that RTC can use via:

`--policy.path=/path/to/checkpoint_dir
`

The main missing pieces are:

generating a proper LeRobot config.json from the OpenPI configs.json,

generating policy_preprocessor / policy_postprocessor configs,

generating the corresponding normalizer / unnormalizer safetensors.

**Comment by khaledmohamed00** [comment]
Workaround: reuse LeRobot’s Pi0 RTC policy artifacts and swap in my model.safetensors

I was able to run examples/rtc/eval_dataset.py with my OpenPI JAX→PyTorch converted Pi0 checkpoint by reusing the non-weight policy artifacts from the LeRobot Pi0 model and replacing only the weights file.

Step 1 — Download a working LeRobot Pi0 policy via the RTC example
python examples/rtc/eval_dataset.py \
  --policy.path=lerobot/pi0_libero_finetuned \
  --dataset.repo_id=HuggingFaceVLA/libero \
  --rtc.execution_horizon=10 \
  --rtc.max_guidance_weight=10.0 \
  --device=cuda


This downloads the policy into the HF cache, e.g.:

~/.cache/huggingface/hub/models--lerobot--pi0_libero_finetuned/

Step 2 — Copy everything except model.safetensors

From the downloaded directory, I copied all policy files except model.safetensors, and then replaced model.safetensors with my converted fine-tuned weights.

Files copied from lerobot/pi0_libero_finetuned:

config.json

policy_preprocessor.json

policy_postprocessor.json

policy_preprocessor_step_5_normalizer_processor.safetensors

policy_postprocessor_step_0_unnormalizer_processor.safetensors

File replaced with my checkpoint:

model.safetensors  (from my OpenPI JAX→PyTorch conversion)

Result

With this directory structure, examples/rtc/eval_dataset.py successfully runs using my fine-tuned weights, because RTC expects a full LeRobot policy directory, not just (model.safetensors + configs.json) from OpenPI.

Note

This workaround assumes that the pre/post-processing + normalizer stats from lerobot/pi0_libero_finetuned are compatible with the checkpoint task/setup. A proper fix would be an official conversion/export path that generates these artifacts from an OpenPI checkpoint.
